frontend/src/components/JoinLeague/JoinLeague.js

```diff
@@ -1,106 +1,150 @@
-          </td>
-        </tr>
-        <tr>
-          <td>League Name 4</td>
-          <td>6/12</td>
-          <td> 13 </td>
-          <td>  
-            <Button type="submit">Join League</Button> 
-          </td>
-        </tr>
-        <tr>
-          <td>League Name 5</td>
-          <td>9/30</td>
-          <td> 12 </td>
-          <td>  
-            <Button type="submit">Join League</Button>
-          </td>
-        </tr>
-      </tbody>
-    </Table>
+    const loadLeagues = async () => {
+        setLeagues(await getLeagues());
+        console.log(leagues);
+    };
 
+    useEffect(() => {
+        loadLeagues();
+    }, []);
 
-<Form.Group controlId="Private Key">
-  <Row>
-    <Form.Label>
-      Private Key:
-    </Form.Label>
-    <Col>
-      <Form.Control type="text" placeholder="Enter private key here" required/>
-      <Form.Control.Feedback type="invalid">
-                        Please provide a Private Key.
-      </Form.Control.Feedback>
-    </Col>
-    <Col>
-      <Button type="submit">Join League</Button>
-    </Col>
-  </Row>
-</Form.Group>
-  </Form>
-</div>
+    return (
+        <div>
+            <Container className="join-league-container">
+                <div className="join-league-form-title">
+                    Join a League
+                </div>
+                {showAlert
+                    && (
+                        <Alert variant="success" onClose={() => setShowAlert(false)} dismissible>
+                            You successfully joined {selectedLeague.leagueName}!
+                        </Alert>
+                    )}
+                {showError
+                    && (
+                        <Alert variant="danger" onClose={() => setShowError(false)} dismissible>
+                            {error}
+                        </Alert>
+                    )}
+                <Form className="join-league-form">
+                    <Table striped bordered variant="light">
+                        <thead>
+                            <tr>
+                                <th>League Name</th>
+                                <th>End Date</th>
+                                <th>Members</th>
+                                <th>Type</th>
+                                <th>Join</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            {leagues && leagues.map((league) => (
+                                <tr>
+                                    <td>{league.leagueName}</td>
+                                    <td>
+                                        {`${new Date(league.settings.endDate).getMonth() + 1
+                                        }/${new Date(league.settings.endDate).getDate() + 1}/${new Date(league.settings.endDate).getFullYear()}`}
+                                    </td>
+                                    <td>{league.playerList.length}</td>
```

Comment:
Can you actually make this

<td>{league.playerList.length} / {league.settings.maxPlayers}</td>

for clarity

Reply:
for line 116

Reply:
done